patches/api/0203-Add-Mob-Goal-API.patch

```diff
@@ -375,7 +375,7 @@ index 0000000000000000000000000000000000000000..659193fc0596084031c09aa47fbb428a
 +    GoalKey<Llama> TRADER_LLAMA_DEFEND_WANDERING_TRADER = GoalKey.of(Llama.class, NamespacedKey.minecraft("trader_llama_defend_wandering_trader"));
 +    GoalKey<Wither> WITHER_DO_NOTHING = GoalKey.of(Wither.class, NamespacedKey.minecraft("wither_do_nothing"));
 +    GoalKey<Illager> RAIDER_OPEN_DOOR = GoalKey.of(Illager.class, NamespacedKey.minecraft("raider_open_door"));
-+    GoalKey<Skeleton> SKELETON_MELEE = GoalKey.of(Skeleton.class, NamespacedKey.minecraft("skeleton_melee"));
++    GoalKey<AbstractSkeleton> SKELETON_MELEE = GoalKey.of(AbstractSkeleton.class, NamespacedKey.minecraft("abstract_skeleton_melee"));
```

Comment:
why rename this? also, shouldn't MobGoalHelper L217 be updated as well?

Reply:
nvm, you did rename it, i swear i double checked. still curious as to why the rename is necessary